### ISSUE TITLE
ML-Benchmark prediction submission: Flow Matching

### ISSUE THREAD

**Issue by mo-tomaswetherell** [WCRP-CORDEX/ml-benchmark#20]
### Institution ID

MO

### Institution Name

Met Office

### Emulator identifier (`emulator_id`)

FlowMatching-v1-orog 

### Emulator Description

Flow matching model, with a U-Net neural network. 

**Predictors**
u, v, q, t, z at multiple pressure levels, normalised with z-score normalisation. 

**Targets**
pr or tasmax; pr normalised with log1p + z-score normalisation, tasmax with z-score normalisation.

**Training configuration**
Adam optimiser.

```
training:
  learning_rate: 0.0001
  epochs: 50
  batch_size: 64
  patience: 10
  years:
    Emulator_hist_future:
      [
        "1961-1966",
        "1968-1974",
        "1976-1980",
        "2080-2086",
        "2088-2094",
        "2096-2099",
      ]
    ESD_pseudo_reality: ["1961-1978"]

validation:
  years:
    Emulator_hist_future: ["1967", "1975", "2087", "2095"]
    ESD_pseudo_reality: ["1979-1980"]
```

**Network configuration**
ADM U-net with:

```
model:
  img_resolution: 128
  model_channels: 128
  channel_mult: [1, 2, 2, 2, 2]
  channel_mult_emb: 2
  num_residual_blocks: 2
  attention_resolutions: [8]
  dropout: 0.1
```

**Flow Matching Hyperparameters**

```
scheduler:
  type: condot

source:
  type: gaussian
  mean: 0.0
  std: 1.0
```

**Other**

Seperate models were trained for predicting precipitation and temperature. Submission includes predictions both with and without orography as a predictor.

### Hardware and Training Details

The models were trained on a single H100 node (94 GB VRAM). 

Each model took ~45 minutes to train (50 epochs, so ~54 seconds per epoch).  

Inference took 2-3 minutes per year of predictions (with a batch size of 256, which used 52 GB of VRAM).

### Stochastic/Probabilistic Output

yes

### Reference URL

https://github.com/mo-tomaswetherell/mo-flow-cordexbench

### Repository for Reproducibility

https://github.com/mo-tomaswetherell/mo-flow-cordexbench

**Comment by jgonzalezab** [comment]
Please fill out [this form](https://forms.gle/1goF329LCrWwdJdh9) so we can give you access to our cluster to transfer your submission.

**Comment by jgonzalezab** [comment]
@mo-tomaswetherell  I’ve been going through your submission and noticed that some prediction files appear to be missing. Specifically, the following files (for both orog and no orog) could not be found:

- `FlowMatching-v1/ALPS_Domain/ESD_pseudo_reality/mid_century/perfect/Predictions_pr_tasmax_MPI-ESM-LR_2041-2060.nc`
- `FlowMatching-v1/NZ_Domain/ESD_pseudo_reality/mid_century/perfect/Predictions_pr_tasmax_EC-Earth3_2041-2060.nc`
- `FlowMatching-v1/SA_Domain/ESD_pseudo_reality/mid_century/perfect/Predictions_pr_tasmax_NorESM2-MM_2041-2060.nc`

Would you mind double-checking whether these files were included? There may potentially be additional missing data, as these files relate to the minimum set of predictions required to run the defined experiments. I’ll wait for your response so we can coordinate the upload of any missing data.

Thanks!

**Comment by mo-tomaswetherell** [comment]
> [@mo-tomaswetherell](https://github.com/mo-tomaswetherell) I’ve been going through your submission and noticed that some prediction files appear to be missing. Specifically, the following files (for both orog and no orog) could not be found:
> 
>     * `FlowMatching-v1/ALPS_Domain/ESD_pseudo_reality/mid_century/perfect/Predictions_pr_tasmax_MPI-ESM-LR_2041-2060.nc`
> 
>     * `FlowMatching-v1/NZ_Domain/ESD_pseudo_reality/mid_century/perfect/Predictions_pr_tasmax_EC-Earth3_2041-2060.nc`
> 
>     * `FlowMatching-v1/SA_Domain/ESD_pseudo_reality/mid_century/perfect/Predictions_pr_tasmax_NorESM2-MM_2041-2060.nc`
> 
> 
> Would you mind double-checking whether these files were included? There may potentially be additional missing data, as these files relate to the minimum set of predictions required to run the defined experiments. I’ll wait for your response so we can coordinate the upload of any missing data.
> 
> Thanks!

Ah, thanks for pointing this out. I double checked and found that I didn't generate these predictions. I'll create them now. 

I think I missed this inference experiment because it's not mentioned in the 'directory structure' section of the [BenchMarkingGuidance](https://docs.google.com/document/d/1zM6Aza1glzBJ8Ewvhu0_0vcm5dvctJ65Bypt1gnN2ck/edit?tab=t.0#heading=h.595v0fw90m99). I realise I should have been referring to the experiments listed in this repo's [README.md](https://github.com/WCRP-CORDEX/ml-benchmark?tab=readme-ov-file). 

Sorry - should be ready by end of today though.

**Comment by jgonzalezab** [comment]
Thanks! Actually, there’s no need to select predictors based on the specific experiments being evaluated. You can simply generate predictions using all predictors available in the `test` folders (see [submission](https://github.com/WCRP-CORDEX/ml-benchmark/tree/main/submission))

**Comment by mo-tomaswetherell** [comment]
> Thanks! Actually, there’s no need to select predictors based on the specific experiments being evaluated. You can simply generate predictions using all predictors available in the `test` folders (see [submission](https://github.com/WCRP-CORDEX/ml-benchmark/tree/main/submission))

Hi Jose -- just finished transfering the missing predictions over. I used the `rsync` command from before. The file is called `FlowMatching-v1-orog-missing-preds.zip`, and it contains both the orography and no orography predictions. Hope that's all good now!
